src/environment.js

```diff
@@ -26,6 +37,66 @@ class Environment {
     return null;
   }
 
+  rawCommitData(commitSha) {
+    // Make sure commitSha is only alphanumeric characters to prevent command injection.
+    if (commitSha.length > 100 || !commitSha.match(/^[0-9a-zA-Z]+$/)) {
```

Comment:
According to Git's [docs on Git Objects](https://git-scm.com/book/id/v2/Git-Internals-Git-Objects), a commit sha is "40-character checksum hash". So probably should max out at that size.

In addition, you could enforce the length into the regex here. Something like this:

```
if (!commitSha.match(/^[0-9a-zA-Z]{0,40}$/)) {
```

Reply:
Yeah, I was aiming to provide a little leeway, but not a lot.  Sometimes the "commitSha" is `HEAD` too.